### ISSUE TITLE
`export_to_pandas` carries a flat ~1.07 s overhead, making interactive use painful

### ISSUE THREAD

**Issue by astigsen** [badoo/pyexasol#372]
**Component:** HTTP transport · `export_to_pandas`
**Version tested:** pyexasol 1.x, Python 3.14, single-node Exasol Personal
**Severity:** medium — documentation or a small-result fast path would resolve it

`export_to_pandas` is the documented way to obtain a DataFrame and the natural call in every
notebook cell. Its cost is dominated by a fixed setup, not by the data:

| rows | `export_to_pandas` | `execute().fetchall()` → DataFrame | ratio |
|---:|---:|---:|---:|
| 100 | 1,078 ms | 6 ms | **180×** |
| 1,000 | 1,076 ms | 7 ms | 154× |
| 3,000 | 1,054 ms | 10 ms | 105× |
| 37,991 | 1,177 ms | — | |
| 120,000 | 1,411 ms | — | |

The overhead is flat: 120,000 rows cost only 31% more than 100. It is HTTP sub-connection setup
(TLS handshake, gzip stream, thread), which pays for itself on a bulk export and is pure loss on
the small result sets that dominate exploratory work.

For scale, the same aggregate computed in-database returns in **5 ms**.

## Reproduction

```python
best = lambda fn, n=3: min(timeit(fn) for _ in range(n))
sql = "SELECT TICKET_ID, BODY FROM BRONZE.TICKETS_RAW LIMIT 100"
best(lambda: C.export_to_pandas(sql))                     # ~1,078 ms
best(lambda: pd.DataFrame(C.execute(sql).fetchall()))     # ~6 ms
```

## Impact

In the evaluation this appeared twice on the critical path: a data scientist's exploration loop,
and an ML engineer's training loop, where **87% of a train-and-score round trip was data movement
rather than modelling**. A notebook doing thirty small pulls spends half a minute waiting on
transport setup.

## Suggested fix

1. Use the ordinary result path below a row/byte threshold and switch to HTTP transport above it —
   the caller should not have to know which regime they are in.
2. Failing that, document the threshold and recommend `fetchall` for exploratory queries. A single
   line in the HTTP-transport guide would recover most of the loss.

**Comment by ArBridgeman** [comment]
@astigsen Thanks for reporting this improvement. We'll discuss & take a look into it.

**Comment by ArBridgeman** [comment]
In our discussions, @tkilias brought up a great point to consider:

> We can't know how many rows we get back. So, we would need to run a query to find out. Which might be more expensive than simply accepting the 1s penalty. We can add guidance to documentation and maybe helper functions.
